core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -124,14 +125,21 @@ public TableMetadata readMetadata() {
   }
 
   ManifestFile writeManifest(DataFile... files) throws IOException {
-    File manifestFile = temp.newFile("input.m0.avro");
+    ManifestEntry[] entries = Arrays.stream(files)
+        .map(file -> manifestEntry(ManifestEntry.Status.ADDED, -1, file))
```

Comment:
This isn't quite correct because for added files, the manifest entry needs to inherit the snapshot ID from the manifest. I know that it will be set to -1, but I think it isn't a good idea to assume it will be here. It also adds an example of creating a manifest entry with an incorrect ID that we don't want people copying.

The alternative is to create two copies of this method, which I think is worth the trade-off.

Reply:
Kept independent.